### DISCUSSION TITLE
"Target class [livewire] does not exist" issue with Laravel package on Laravel 9

### DISCUSSION THREAD

**Discussion by Okipa** [livewire/livewire#5117] [1. Help] [answered]
Hi there,

I'm developing a new version of a package that is now based on Livewire and I'm stuck on a CI issue related to Livewire and Laravel 9.  
It drives me crazy as I have no clue about how to fix this. I've made lot a research and unfortunately, nothing jumped out of it.  
The new version is ready to be released in stable but this CI issue is now blocking me to tag the package.  
It has to ben ran on PHP 8.1 and is compatible with Laravel 8 and 9.

The error that is displayed is `Target class [livewire] does not exist` and only occurs when i'm launching PHPStan (Larastan) and PHPUnit with L9, everything is going fine with L8, even when I'm installing the package and manually testing the features on a real app.

Would you have an idea of what could be going on here?

Here is the package version : https://github.com/Okipa/laravel-table/tree/v5

You'll be able to see this error here on PHPUnit here: https://github.com/Okipa/laravel-table/runs/7448675623?check_suite_focus=true#step:8:12 

Executing `vendor/bin/phpstan analyse --memory-limit=2G` locally is triggering the error locally and on CI.

**Comment by joshhanley** [comment]
@Okipa hmm that is strange. Did you say you can replicate the issue in a sample Laravel 9 app? If so can you share a repo to that so we can investigate?

Thanks!

**Reply by Okipa** [reply]
@joshhanley Sorry for the delayed answer!

No, it is not replicable with the package installed and running in an app, the issue is only appearing when running PHPStan locally and on CI (Github action).

To check the issue, you just have to clone the package https://github.com/Okipa/laravel-table, switch on `V5` branch, install dependencies with a `composer install` and run `composer phpstan`, or `vendor/bin/phpstan analyse --memory-limit=2G`.

I honestly have no clue on how to resolve this.

**Reply by Okipa** [reply]
Hi @joshhanley, sorry for seeking help one more time.

The package v5 is ready to be released but I'm not really confident about releasing it without an operational PHPStan analysis.

Could you take a look on this ?

Do you thing I should open an issue on the PHPStan or the Larastan repo instead of here ?

**Reply by luanfreitasdev** [reply]
Hello @Okipa !. I think the problem is this: (I just don't know the right reason). but it makes sense

```php
if (! $this->app->runningInConsole()) {
     Livewire::component('table', \Okipa\LaravelTable\Livewire\Table::class);
}
```
https://github.com/Okipa/laravel-table/blob/v5/src/LaravelTableServiceProvider.php#L31

![image](https://user-images.githubusercontent.com/33601626/187090903-a49bd770-2e2e-4395-8aa7-7a259e59a632.png)

Note: in CI you will have to choose to run phpstan only in laravel 9 because of the incompatibility between L8 and L9:

* https://github.com/Power-Components/livewire-powergrid/blob/main/.github/workflows/php.yml#L82
* https://github.com/Power-Components/livewire-powergrid/blob/main/larastan.sh#L3

**Reply by Okipa** [reply]
Thanks @luanfreitasdev for your answer!

Adding this condition allows PHPStan to run and to display the analysis errors but the issue still happens at the end of the process.

I had another clue about what could be going wrong here: https://github.com/nunomaduro/larastan/discussions/1345#discussioncomment-3493026

I'm waiting for the answer of canvural to see if this is something that should be fixed in Larastan or if there is a way to avoid the issue by declaring something somewhere 🙃

**Reply by luanfreitasdev** [reply]
That same error? I ran it here locally and it seems to work without any problems, just syntax

**Reply by Okipa** [reply]
Yes, see below.

In fact, adding your condition will fail in tests because they are ran in console, so it seems that it is not a suitable solution...

![Capture d’écran 2022-08-29 à 13 38 35](https://user-images.githubusercontent.com/5328934/187193233-abe34df3-d135-47db-bce3-408dc5b246db.png)

**Comment by Okipa** [comment, marked as answer]
Sorry for the late update => This has been fixed in https://github.com/nunomaduro/larastan/releases/tag/2.2.3.

**Reply by joshhanley** [reply]
@Okipa thanks for the update! 🙂
